README.md

```diff
@@ -1,28 +1,78 @@
 # github-action-detect-unused-secrets
 
-GitHub Action that automatically detects unused secrets comparing the workflow secret names with the repo secret names.
+GitHub Action that detects unused secrets.
+Compares the secret names in your repository with the secret names under `.github/workflows`
 
 ## Example usage
 
-```yaml
-name: detect-unused-secrets
+The example below runs the action every Monday at 10am.
+Generates a token with required permissions which `secrets.GITHUB_TOKEN` does not have.
+Uses checkout code to get access to the `$GITHUB_WORKSPACE`,
+And finally runs the `github-action-detect-unused-secrets` with the generated token.
 
-on: [push]
+```yml
+ name: detect-unused-secrets
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 10 * * MON'
 
 jobs:
   setup:
     runs-on: ubuntu-latest
     steps:
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@v1
+        with:
+          app_id: ${{ secrets.APP_ID }}
+          private_key: ${{ secrets.PRIVATE_KEY }}
+      - name: Checkout Code
+      - uses: actions/checkout@v3
       - name: Detect unused secrets
-        uses: nearform/github-action-detect-unused-secrets@v1
+        uses: nearform/github-action-detect-unused-secrets@main
         with:
-          github-token: ${{ secrets.GITHUB_TOKEN }}
+          github-token: ${{ steps.generate_token.outputs.token }}
+
 ```
 
 ## Usage
 
 Configure this action in your workflows providing the inputs described below.
 
+### Install a GitHub App
+
+**Required** [Create a GitHub App](https://docs.github.com/en/developers/apps/building-github-apps/creating-a-github-app) within your organisation with permission to `secrets: read-only` and [install](https://docs.github.com/en/developers/apps/managing-github-apps/installing-github-apps#installing-your-private-github-app-on-your-repository) it on the repo where you want to use this action. Save the `APP_ID` and generated `PRIVATE_KEY` to the secrets under your organisation or repo.
+
+This action uses the [listRepoSecrets endpoint](https://docs.github.com/en/rest/reference/actions#list-repository-secrets), your GitHub App must have the `secrets` repository permission to use this endpoint.
+
+> Note that you can skip this step if your repo is `not` under an organisation, you can [create a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) `with repo scope` and reference it as github-token
```

Comment:
this is not accurate. you can use a PAT if you prefer, it doesn't matter if the repo is under an org or a personal account. it's just our decision to not use PATs for our org